source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/MediaRenderer.java

```diff
@@ -206,18 +234,103 @@ private ImageView renderPlayButton(
         return playButtonView;
     }
 
-    private FullscreenVideoView renderMediaPlayer(
-            Context context,
-            ViewGroup viewGroup,
-            Media media,
-            HostConfig hostConfig)
+    static class ControllerVisibilityListener implements StyledPlayerView.ControllerVisibilityListener
+    {
+        @Override
+        public void onVisibilityChanged(int visibility) {
+
+        }
+    }
+
+    class PlayerListener implements Player.Listener
+    {
+        private MediaOnCompletionListener m_onCompletionListener = null;
+
+        public PlayerListener(MediaOnCompletionListener onCompletionListener)
+        {
+            m_onCompletionListener = onCompletionListener;
+        }
+
+        @Override
+        public void onEvents(Player player, Player.Events events)
+        {
+            if (events.contains(EVENT_PLAYBACK_STATE_CHANGED))
+            {
+                m_onCompletionListener.onCompletion(null);
+            }
+        }
+    }
+
+    private View renderNewMediaPlayer(
+        RenderedAdaptiveCard renderedCard,
+        Context context,
+        ViewGroup viewGroup,
+        Media media,
+        ICardActionHandler cardActionHandler,
+        HostConfig hostConfig)
+    {
+        StyledPlayerView playerView = new StyledPlayerView(context);
+        playerView.setUseController(true);
+        playerView.setControllerHideOnTouch(true);
+        playerView.setResizeMode(AspectRatioFrameLayout.RESIZE_MODE_FIXED_WIDTH);
+        playerView.setControllerVisibilityListener(new ControllerVisibilityListener());
```

Comment:
If ControllerVisibilityListener is used only once, I prefer anonymous function

```
playerView.setControllerVisibilityListener(new StyledPlayerView.ControllerVisibilityListener() {
@Override
        public void onVisibilityChanged(int visibility) {
 // Add why you have empty body here
       }
});

```

Reply:
class has been removed